streams/src/main/java/org/apache/kafka/streams/processor/internals/KafkaStreamingPartitionAssignor.java

```diff
@@ -63,38 +75,95 @@ public String name() {
 
     @Override
     public Subscription subscription(Set<String> topics) {
-        return new Subscription(new ArrayList<>(topics));
+        Set<TaskId> prevTasks = streamThread.prevTasks();
+        Set<TaskId> standbyTasks = streamThread.cachedTasks();
+        standbyTasks.removeAll(prevTasks);
```

Comment:
Maybe we can leave some comments in here and in the assignor's logic with the descriptions in the PR. For example, under which scenario would a client have some checkpoints of tasks that have already been migrated out of this client.